Data race in tests

Observed in my workstation Linux amd64, with all versions of Go1.13+
```
$ go test -race github.com/evanw/esbuild/internal/bundler
==================
WARNING: DATA RACE
Read at 0x00c0001247e0 by goroutine 20:
  github.com/evanw/esbuild/internal/bundler.(*scanner).preprocessInjectedFiles.func2()
      esbuild/internal/bundler/bundler.go:1085 +0x75

Previous write at 0x00c0001247e0 by goroutine 63:
  github.com/evanw/esbuild/internal/bundler.(*scanner).preprocessInjectedFiles()
      esbuild/internal/bundler/bundler.go:1078 +0x13f7
  github.com/evanw/esbuild/internal/bundler.ScanBundle()
      esbuild/internal/bundler/bundler.go:891 +0x4f6
  github.com/evanw/esbuild/internal/bundler.(*suite).expectBundled.func1()
      esbuild/internal/bundler/bundler_test.go:96 +0x597
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1194 +0x202

Goroutine 20 (running) created at:
  github.com/evanw/esbuild/internal/bundler.(*scanner).preprocessInjectedFiles()
      esbuild/internal/bundler/bundler.go:1084 +0x15b1
  github.com/evanw/esbuild/internal/bundler.ScanBundle()
      esbuild/internal/bundler/bundler.go:891 +0x4f6
  github.com/evanw/esbuild/internal/bundler.(*suite).expectBundled.func1()
      esbuild/internal/bundler/bundler_test.go:96 +0x597
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1194 +0x202

Goroutine 63 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1239 +0x5d7
  github.com/evanw/esbuild/internal/bundler.(*suite).expectBundled()
      esbuild/internal/bundler/bundler_test.go:84 +0x1ae
  github.com/evanw/esbuild/internal/bundler.TestInject()
      esbuild/internal/bundler/bundler_default_test.go:3393 +0x8bc
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1194 +0x202
==================
--- FAIL: TestInject (0.00s)
    --- FAIL: TestInject/#00 (0.00s)
        testing.go:1093: race detected during execution of test
    testing.go:1093: race detected during execution of test
FAIL
FAIL	github.com/evanw/esbuild/internal/bundler	1.317s
FAIL
```

--- a/internal/bundler/bundler.go
+++ b/internal/bundler/bundler.go
@@ -1057,6 +1057,8 @@ func (s *scanner) preprocessInjectedFiles() {
 		go func() { s.resultChannel <- result }()
 	}
 
+	results := make([]config.InjectedFile, len(s.options.InjectAbsPaths))
+	j := 0
 	for _, absPath := range s.options.InjectAbsPaths {
 		prettyPath := s.res.PrettyPath(logger.Path{Text: absPath, Namespace: "file"})
 		lowerAbsPath := lowerCaseAbsPathForWindows(absPath)
@@ -1074,20 +1076,22 @@ func (s *scanner) preprocessInjectedFiles() {
 			continue
 		}
 
-		i := len(injectedFiles)
-		injectedFiles = append(injectedFiles, config.InjectedFile{})
 		channel := make(chan config.InjectedFile)
 		s.maybeParseFile(*resolveResult, prettyPath, nil, logger.Range{}, nil, inputKindNormal, channel)
 
 		// Wait for the results in parallel
+		// The results slice is large enough, it is not reallocated during the computations
 		injectWaitGroup.Add(1)
-		go func(i int, prettyPath string, resolveResult *resolver.ResolveResult) {
-			injectedFiles[i] = <-channel
+		go func(i int) {
+			results[i] = <-channel
 			injectWaitGroup.Done()
-		}(i, prettyPath, resolveResult)
+		}(j)
+		j++
 	}
 
 	injectWaitGroup.Wait()
+	injectedFiles = append(injectedFiles, results[:j]...)
+
 	s.options.InjectedFiles = injectedFiles
 }
 

